### PR TITLE
feat: allow promoting an existing template version to active from CLI

### LINKED ISSUE #15042: Allow promoting an existing template version to active from CLI
Coder allows promoting a version to active from the Dashboard and also through API but this functionality is missing from the CLI,

`coder templates versions` do not expose a command to achieve that.

`coder templates versions list kasmvnc-desktop`
```console
NAME                 CREATED AT            CREATED BY  STATUS     ACTIVE
amazing_hermann5     2024-10-04T11:56:34Z  matifali    Failed
bold_colden0         2024-10-04T13:16:39Z  matifali    Succeeded
dazzling_germain1    2024-10-04T13:02:40Z  matifali    Succeeded
distracted_yonath9   2024-10-04T13:28:19Z  matifali    Succeeded
elegant_archimedes9  2024-10-04T11:02:17Z  matifali    Succeeded
fervent_khayyam4     2024-10-04T12:00:07Z  matifali    Succeeded
hardcore_merkle3     2024-10-04T11:53:59Z  matifali    Failed
kind_goldstine6      2024-10-04T11:52:43Z  matifali    Failed
nice_wozniak0        2024-10-04T11:06:50Z  matifali    Succeeded
romantic_merkle9     2024-10-04T14:16:13Z  matifali    Succeeded  Active
sleepy_robinson1     2024-10-04T11:56:06Z  matifali    Failed
tender_ritchie1      2024-10-04T11:58:26Z  matifali    Succeeded
```

`coder templates versions` gives,

```console
coder v2.16.0+683a720

USAGE:
  coder templates versions

  Manage different versions of the specified template

  Aliases: version

    - List versions of a specific template:

        $ coder templates versions list my-template


SUBCOMMANDS:
    archive      Archive a template version(s).
    list         List all the versions of the specified template
    unarchive    Unarchive a template version(s).

———
Run `coder --help` for a list of global options.
```

## Suggestion

Add a promote sub-command like `coder templates versions promote -t=<TEMPLATE_NAME> --tempplate-version=<TEMPLATE_VERSION_NAME>`
Here `<TEMPLATE_NAME>` is the name of the template and `<TEMPLATE_VERSION_NAME>` is the name of the template version.

This is related to #14860 and will help solve #9899

### PR DESCRIPTION
**Pull request by joobisb**
This PR closes https://github.com/coder/coder/issues/15042

### CODE REVIEW

File: cli/templateversions.go [hunk 1]
```diff
@@ -169,3 +170,66 @@ func templateVersionsToRows(activeVersionID uuid.UUID, templateVersions ...coder
...
+
+func (r *RootCmd) templateVersionsPromote() *serpent.Command {
+	var (
+		templateName        string
+		templateVersionName string
+		orgContext          = NewOrganizationContext()
+	)
+	client := new(codersdk.Client)
+	cmd := &serpent.Command{
+		Use:   "promote",
```
Comment (Emyrk) [inline cli/templateversions.go:182]: I don't think there is precedence for required flags, can we add to the `Use` to include the required flags?

```suggestion
		Use:   "promote --template=<template_name> --template-version=<template_version_name>",
```

We include required params in other `Use` fields: https://github.com/coder/coder/blob/61c37d039bd5f9a3fac005502d36f436a7ce2e24/cli/autoupdate.go#L18-L18

File: cli/testdata/coder_templates_versions_promote_--help.golden [hunk 2]
```diff
@@ -0,0 +1,22 @@
+coder v0.0.0-devel
+
+USAGE:
+  coder templates versions promote [flags]
```
Comment (Emyrk) [inline cli/testdata/coder_templates_versions_promote_--help.golden:4]: I know the output `[flags]` technically includes my usage comment earlier. Given they are required, I wonder if we should include them more explicitly.

**Comment by joobisb** [comment]
@Emyrk I've addressed the comments, please do have a look

**Review by Emyrk — APPROVED** [review]

**Comment by joobisb** [comment]
@Emyrk I think CI failure is unrelated, can we merge this?

**Comment by Emyrk** [comment]
It was, thanks @matifali :+1:

**Pull request merged**